src/app/services/mode-of-use.service.ts

```diff
@@ -0,0 +1,45 @@
+let modeOfUseServiceIsInstantiated: boolean;
+
+class ModeOfUseService {
+
+	constructor() {
+		if (modeOfUseServiceIsInstantiated) {
+			throw new Error('Le routeur est déjà instancié.');
+		}
+
+		modeOfUseServiceIsInstantiated = true;
+	}
+
+	setSelectedMode(newSelectedMode: string): void {
+		localStorageServiceInstance.getItem('modeOfUse').then((result: any) => {
+			let json = result;
+			json.selectedMode = newSelectedMode;
+			localStorageServiceInstance.setItem('modeOfUse', json);
+		});
+	}
+
+	setSettingValue(key: string, newIndex: number): Promise<boolean> {
+		let jsonIsEdited = false;
+		return localStorageServiceInstance.getItem('modeOfUse')
```

Comment:
En voyant cette adhérence entre les deux services, je me demande si on ne devrait pas déplacer ces deux méthodes dans le `localStorageService` finalement… Un avis ?

Reply:
Haha c'est ce que j'avais fait au début et j'ai changé au dernier moment.

En fait j'ai changé d'avis car vu que les deux méthodes sont identiques pour le serveur et l'extension, et que c'est propre à un fichier (mode-of-use), je me suis dit qu'il fallait un fichier dédié.

Reply:
C’est pas déconnant, ça évite la duplication du code. Mon cœur balance, du coup :D

Reply:
Après délibération avec l'équipe nous allons conserver un fichier distinct pour la maintenabilité et pour éviter la duplication de code.